flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/GenericArraySerializerSnapshot.java

```diff
@@ -76,10 +80,26 @@ protected void readOuterSnapshot(
         this.componentClass = InstantiationUtil.resolveClassByName(in, userCodeClassLoader);
     }
 
+    @Override
+    public TypeSerializerSchemaCompatibility<C[]> resolveSchemaCompatibility(
+            TypeSerializerSnapshot<C[]> oldSerializerSnapshot) {
+        if (oldSerializerSnapshot instanceof GenericArraySerializerConfigSnapshot) {
```

Comment:
imo, there is something wrong with this branch. If the `oldSerializerSnapshot` is `GenericArraySerializerConfigSnapshot` we then call:
`CompositeTypeSerializerUtil.delegateCompatibilityCheckToNewSnapshot` -> `CompositeTypeSerializerSnapshot#internalResolveSchemaCompatibility` ->
`GenericArraySerializerSnapshot#resolveOuterSchemaCompatibility` where we try to cast the initial
`GenericArraySerializerConfigSnapshot` to `GenericArraySerializerSnapshot` which will fail.

Could you please verify that, preferably with a test? Or if it works point me to a test that checks that?

Reply:
You are right.
I have added CompositeTypeSerializerUpgradeTest#testUpgradeFromDeprecatedSnapshot to test it.
BTW, I have a minor problem. 
TypeSerializerUpgradeTestBase seems just to test upgradation from Flink Version 1.11 ?
IIUC, TypeSerializer should supports Flink Version from 1.8 currently.
Should we need to make it test from 1.8 ?